pkg/provision/digitalocean.go

```diff
@@ -54,12 +55,72 @@ func (p *DigitalOceanProvisioner) Status(id string) (*ProvisionedHost, error) {
 	}, nil
 }
 
-func (p *DigitalOceanProvisioner) Delete(id string) error {
-	sid, _ := strconv.Atoi(id)
-	_, err := p.client.Droplets.Delete(context.Background(), sid)
+// Delete terminates the exit node
+func (p *DigitalOceanProvisioner) Delete(config HostDeleteConfig) error {
+	var id string
+	var err error
+	if len(config.IP) > 0 {
+		id, err = p.lookupInstance(config)
+		if err != nil {
+			return err
+		}
+	} else {
+		id = config.ID
+	}
+	sid, err := strconv.Atoi(id)
+	if err != nil {
+		return err
+	}
+	_, err = p.client.Droplets.Delete(context.Background(), sid)
 	return err
 }
 
+// List returns a list of exit nodes
+func (p *DigitalOceanProvisioner) List(filter ListFilter) ([]*ProvisionedHost, error) {
+	var inlets []*ProvisionedHost
+	opt := &godo.ListOptions{}
+	for {
+		droplets, resp, err := p.client.Droplets.ListByTag(context.Background(), filter.Filter, opt)
+		if err != nil {
+			return inlets, err
+		}
+		for _, droplet := range droplets {
+			publicIP, err := droplet.PublicIPv4()
+			if err != nil {
+				return inlets, err
+			}
+			host := &ProvisionedHost{
+				IP: publicIP,
+				ID: fmt.Sprintf("%d", droplet.ID),
+			}
+			inlets = append(inlets, host)
+		}
+		if resp.Links == nil || resp.Links.IsLastPage() {
+			break
+		}
+		page, err := resp.Links.CurrentPage()
+		if err != nil {
+			return inlets, err
+		}
+		opt.Page = page + 1
+	}
+	return inlets, nil
+}
+
+func (p *DigitalOceanProvisioner) lookupInstance(config HostDeleteConfig) (string, error) {
+	inlets, err := p.List(ListFilter{Filter: "inlets"})
+	if err != nil {
+		return "", err
+	}
+	for _, inlet := range inlets {
+		if inlet.IP == config.IP {
+			return inlet.ID, nil
+		}
+	}
+	return "", fmt.Errorf("No host with ip: %s", config.IP)
```

Comment:
Go errors AFAIK should be lowercase for the first word

Reply:
Yes, there are a few instances of this in other parts of the code, are you happy for me to change them at the same time?